src/sentry/workflow_engine/endpoints/organization_detector_details.py

```diff
@@ -169,6 +172,7 @@ def delete(self, request: Request, organization: Organization, detector: Detecto
             return Response(status=403)
 
         RegionScheduledDeletion.schedule(detector, days=0, actor=request.user)
+        detector.update(status=ObjectStatus.PENDING_DELETION)
```

Comment:
Setting it to pending deletion in the delete endpoint after it's been scheduled for deletion